compiler/rustc_const_eval/src/util/check_validity_requirement.rs

```diff
@@ -53,7 +53,10 @@ fn check_validity_requirement_strict<'tcx>(
     let mut cx = InterpCx::new(cx.tcx(), DUMMY_SP, cx.typing_env, machine);
 
     let allocated = cx
-        .allocate(ty, MemoryKind::Machine(crate::const_eval::MemoryKind::Heap))
+        .allocate(
+            ty,
+            MemoryKind::Machine(crate::const_eval::MemoryKind::Heap { was_made_immut: false }),
```

Comment:
Wtf why does this use "heap"? That makes no sense.^^
I understand this is pre-existing, but please at least add a FIXME. Or just make it `Stack` which is at least less wrong.